File: java/0014_Longest_Common_Prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0) {
            return "";
        }
        String prefix = strs[0];
        for (int i = 1; i < strs.length; i++) {
            String temp = prefix;
            char[] chars = strs[i].toCharArray();
            int len = Math.min(temp.length(), chars.length);
            int j = 0;
            for (; j < len; j++) {
                if (temp.charAt(j) != chars[j]) {
                    break;
                }
            }
            prefix = (j > 0) ? temp.substring(0, j) : "";
        }
        return prefix;
    }
}

File: kotlin/0014_Longest_Common_Prefix.kt
class Solution {
    fun longestCommonPrefix(strs: Array<String>): String {
        if (strs.isEmpty()) {
            return ""
        }
        strs[0].forEachIndexed { i, ch ->
            for (j in 1 until strs.size) {
                if (i >= strs[j].length || ch != strs[j][i]) {
                    return strs[0].substring(0 until i)
                }
            }
        }
        return strs[0]
    }
}